src/Models/LoginModel.php

```diff
@@ -28,7 +29,8 @@ class LoginModel extends Model
     protected $useTimestamps   = false;
     protected $validationRules = [
         'ip_address' => 'required',
-        'identifier' => 'required',
+        'id_type'    => 'required',
+        'identifier' => 'permit_empty|string',
```

Comment:
When can this be empty?

Reply:
When failed login, it may be empty.

Reply:
Would that even get as far as a login attempt? I would think basic validation errors (like entering `1234` in an email field) should not be recorded as "attempts", especially since they won't ever link to an actual user. In this case I would be in favor of always requiring the identifier field.

Reply:
It is a failed attempt. I don't know why it should not be recorded.
Failed attempts can't link to an actual user.

Reply:
I won't push it.